File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/project_settings/modules/ModuleSourceSettings.java
package com.github.alexandrecarlton.idea.settings.layout.project_settings.modules;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import org.immutables.value.Value;

import java.nio.file.Path;
import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableModuleSourceSettings.class)
public interface ModuleSourceSettings {

  Path contentRoot();

  List<String> sources();

  List<String> tests();

  List<String> resources();

  List<String> testResources();

  List<String> excluded();

}

File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/project_settings/modules/ModuleSourceSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.project_settings.modules

import java.nio.file.Path

data class ModuleSourceSettings(

    val contentRoot: Path,

    val sources: List<String>? = null,

    val tests: List<String>? = null,

    val resources: List<String>? = null,

    val testResources: List<String>? = null,

    val excluded: List<String>? = null
)
